### PR TITLE
Update golang.org/x/net/... dependencies to release-branch.go1.12

### LINKED ISSUE #70819: kube-controller-manager becomes deadlocked but still passes healthcheck
**What happened**:

All three controller managers "wedged" in a high availability cluster--zero network traffic between any of the controller managers and the api server--due to a deadlock.  All three controller manager processes running and passing healthcheck, but none performing any controller manager duties.

pprof stackdump surfaced a deadlock in the golang.org/x/net/http2 library on all masters preventing any http2 client requests from being made from the controller managers:

One goroutine already holds the `clientConnPool.mu` lock and is deadlocked attempting to acquire the `ClientConn.mu` lock:

```
goroutine 1776 [semacquire, 3556 minutes]:
sync.runtime_SemacquireMutex(0xc420970a14, 0x522100)
        /usr/local/go/src/runtime/sema.go:71 +0x3d
sync.(*Mutex).Lock(0xc420970a10)
        /usr/local/go/src/sync/mutex.go:134 +0xee
k8s.io/kubernetes/vendor/golang.org/x/net/http2.(*ClientConn).CanTakeNewRequest(0xc4209709c0, 0xc420929500)
        /workspace/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/golang.org/x/net/http2/transport.go:611 +0x3f
k8s.io/kubernetes/vendor/golang.org/x/net/http2.(*clientConnPool).getClientConn(0xc420985560, 0xc4295eeb00, 0xc429692cc0, 0xd, 0xc4207d9800, 0xc426376f68, 0x94bb20, 0xc4207d9810)
        /workspace/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/golang.org/x/net/http2/client_conn_pool.go:67 +0xe8
```

Another already holds the `ClientConn.mu` lock and is deadlocked attempting to acquire the `clientConnPool.mu` lock:

```
goroutine 85 [semacquire, 3556 minutes]:
sync.runtime_SemacquireMutex(0xc42098556c, 0x528300)
        /usr/local/go/src/runtime/sema.go:71 +0x3d
sync.(*Mutex).Lock(0xc420985568)
        /usr/local/go/src/sync/mutex.go:134 +0xee
k8s.io/kubernetes/vendor/golang.org/x/net/http2.(*clientConnPool).MarkDead(0xc420985560, 0xc4209709c0)
        /workspace/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/golang.org/x/net/http2/client_conn_pool.go:201 +0x4b
k8s.io/kubernetes/vendor/golang.org/x/net/http2.noDialClientConnPool.MarkDead(0xc420985560, 0xc4209709c0)
        <autogenerated>:1 +0x3e
panic(0x3058280, 0x3d35e60)
        /usr/local/go/src/runtime/panic.go:491 +0x283
k8s.io/kubernetes/vendor/golang.org/x/net/http2.(*pipe).closeWithError(0xc42451ea28, 0xc42451ea78, 0x0, 0x0, 0x0)
        /workspace/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/golang.org/x/net/http2/pipe.go:106 +0x214
k8s.io/kubernetes/vendor/golang.org/x/net/http2.(*pipe).CloseWithError(0xc42451ea28, 0x0, 0x0)
        /workspace/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/golang.org/x/net/http2/pipe.go:93 +0x53
k8s.io/kubernetes/vendor/golang.org/x/net/http2.(*clientConnReadLoop).cleanup(0xc420a51fb0)
        /workspace/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/golang.org/x/net/http2/transport.go:1408 +0x219
```
**What you expected to happen**:

kube-controller-manager does not become deadlocked

kube-controller-manager should fail healthcheck when in a broken state like this. In this case, if the kube-controller-manager were to fail healthcheck when not able to reach the kube-apiserver for it's regular leader lock acquire attempts for some time threshold, the issue would have been mitigated by restarts.

**How to reproduce it (as minimally and precisely as possible)**:

Exact steps to reproduce state not yet known. We found the cluster in the state and examined it in situ and were able to capture a pprof stackdump which proves the existence of the deadlock. We believe it may be possible to recreate the http2 error state that triggers deadlock with some careful analysis of the deadlock, but have not gotten that far yet.

**Anything else we need to know?**:

The call from `http2.(*clientConnPool).getClientConn() `to `http2.(*CLientConn).CanTakeNewRequest()` [was removed](https://go-review.googlesource.com/c/net/+/122590/) from the call path by https://github.com/golang/net/commit/6a8eb5e2b1816b30aa88d7e3ecf9eb7c4559d9e6 and is available in the `release-branch.go1.11` branch of `x/net/http2`. Bumping the vendored version of `x/net/http2` to `6a8eb5e2b1816b30aa88d7e3ecf9eb7c4559d9e6` or higher should resolve the deadlock issue.

**Environment**:
- Kubernetes version: 1.10.6

/sig api-machinery
cc @cheftako @mml @wenjiaswe @lavalamp 

/kind bug

### LINKED ISSUE #71016: Update golang.org/x/net/http2 vendored files
It is too late to do this in 1.13, so when 1.14 opens up, we should update `golang.org/x/net/http2`

Please see https://github.com/kubernetes/kubernetes/pull/70971

### PR DESCRIPTION
**Pull request by dims**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Updates vendored files in golang.org/x/net/... packages to release-branch.go1.12

We have been trying to update x/net for a while, for example in https://github.com/kubernetes/kubernetes/pull/70971 and see https://github.com/kubernetes/kubernetes/pull/74781 on why we need the new packages. So in this PR, we just update the x/net package and then other folks can file PRs for what they see like 74781. This problem with x/net has stopped us from updating things like our docker/docker dependency (as docker is already using latest x/net). So this fixes some issues and allows us to move forward on other areas mentioned earlier.

**Which issue(s) this PR fixes**:

Fixes #71016
Fixes #70819

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -2385,8 +2388,8 @@
 		},
 		{
 			"ImportPath": "github.com/grpc-ecosystem/go-grpc-middleware",
-			"Comment": "v1.0.0-4-g498ae206fc3cfe",
-			"Rev": "498ae206fc3cfe81cd82e48c1d4354026fa5f9ec"
+			"Comment": "v1.0.0-8-gcfaf5686ec79ff",
+			"Rev": "cfaf5686ec79ff8344257723b6f5ba1ae0ffeb4d"
```
Comment (liggitt) [inline Godeps/Godeps.json]: this was surprising... is it spurious?

**Comment by liggitt** [comment]
/milestone v1.14

replacement for https://github.com/kubernetes/kubernetes/pull/71501

**Comment by dims** [comment]
So thinking about the `hack` in the version of `godep` ... namely the `-i` parameter introduced here.

1) If we did not have the hack we would not be able to use x/net until all our dependencies move over, so that does not work.
2) If someone uses the Godeps/Godeps.json we generate, will it work with stock `godep`, yes, it will work, but the x/net they end up pulling in may not be what we specify in our Godeps/Godeps.json
3) If someone wants to use the x/net, they will have to use our forked version of `godep` and will have to explicitly specify the newer x/net in the godep command line like we do.

@liggitt WDYT?

**Comment by liggitt** [comment]
> 1. If we did not have the hack we would not be able to use x/net until all our dependencies move over, so that does not work.

what is the failure mode? does `godep save` fail or does it pull in the wrong version?

**Comment by dims** [comment]
`godep save` pulls wrong version

**Comment by liggitt** [comment]
if we can pull in the version we want that corresponds to go1.12, and publish that version in our Godeps.json, and downstream use of Godep at worst pulls in a version we don't want (based on conflicting requirements from the dependency tree), that seems reasonable short-term (1.14 timeframe).

Tools like `go mod`, dep, and glide can pin specific versions to use, and downstream godep users can do things like what we did.

In the 1.15 timeframe, I'd like to see [progress toward go modules](https://github.com/kubernetes/kubernetes/pull/74877) and removal of the Godep vendor management.

**Comment by dims** [comment]
Correct. the PR reflects `if we can pull in the version we want that corresponds to go1.12, and publish that version in our Godeps.json`

**Comment by liggitt** [comment]
modulo the spurious grpc sha change, this lgtm

**Comment by dims** [comment]
@liggitt the grpc SHA was because we convinced them to move to newer x/net please see the PR https://github.com/grpc-ecosystem/go-grpc-middleware/pull/191

**Comment by liggitt** [comment]
ah, great, thanks

**Comment by dims** [comment]
For the record (someone stumbling here in the future) see the other attempts/history 
- https://github.com/kubernetes/kubernetes/pull/71501
- https://github.com/kubernetes/kubernetes/pull/74466
- https://github.com/kubernetes/kubernetes/pull/74460

especially the list of dependencies that use old x/net:
- https://github.com/kubernetes/kubernetes/pull/71501#issuecomment-466739003

**Comment by dims** [comment]
@thockin WDYT?

**Comment by liggitt** [comment]
/lgtm
/approve

/hold
for @thockin's ack

**Comment by dims** [comment]
The other way to look at it is the `Godeps.json` is correct, it reflects exactly what we ship and want people to use :)

**Comment by dims** [comment]
@thockin can you please remove hold on this if you are ok with it?

**Comment by wojtek-t** [comment]
/lgtm

Thanks a lot for doing that @dims !

**Comment by dims** [comment]
so in the interest of getting more soak time, i'll remove the hold

/hold cancel

**Comment by wojtek-t** [comment]
Awesome - thanks a lot @dims !

**Comment by wenjiaswe** [comment]
Thanks @dims !

**Pull request merged**